x/cdp/keeper/interest.go

```diff
@@ -98,6 +159,21 @@ func (k Keeper) UpdateFeesForAllCdps(ctx sdk.Context, collateralType string) err
 	return iterationErr
 }
 
+// CalculateFees returns the fees accumulated since fees were last calculated based on
+// the input amount of outstanding debt (principal) and the number of periods (seconds) that have passed
+func (k Keeper) CalculateFees(ctx sdk.Context, principal sdk.Coin, periods sdk.Int, collateralType string) sdk.Coin {
+	// how fees are calculated:
+	// feesAccumulated = (outstandingDebt * (feeRate^periods)) - outstandingDebt
+	// Note that since we can't do x^y using sdk.Decimal, we are converting to int and using RelativePow
+	feePerSecond := k.getFeeRate(ctx, collateralType)
+	scalar := sdk.NewInt(1000000000000000000)
```

Comment:
Use the `scalingFactor` var declared on line 13.

Reply:
80651c3